### PR TITLE
Implements GET Request of inventory, returning JSON

### PR DESCRIPTION
**Pull request by JulesRoadknight**

### CODE REVIEW

File: src/main/java/com/example/gildedrose/GildedRoseService.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.example.gildedrose;
+
+import com.example.gildedrose.item.*;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+@Service
+public class GildedRoseService {
```
Comment (damonkelley) [inline src/main/java/com/example/gildedrose/GildedRoseService.java]: **praise**: Nice job keeping Spring annotations from spreading throughout the application!

File: src/main/java/com/example/gildedrose/GildedRoseService.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    private GildedRose gildedRose;
+
+    public GildedRoseService() {
+        this.gildedRose = new GildedRose(new ArrayList<>(Arrays.asList(
```
Comment (damonkelley) [inline src/main/java/com/example/gildedrose/GildedRoseService.java]: **question**: Do we need the `GildedRose` for this service?

We are only using the new `getItems`. It feels like we might be violating one or more a SOLID princples.

File: src/main/java/com/example/gildedrose/item/GildedRose.java [hunk 3]
```diff
@@ -12,4 +12,8 @@ public GildedRose(ArrayList<Updateable> items) {
     public void updateQuality() {
         items.forEach(Updateable::update);
     }
+
+    public ArrayList<Updateable> getItems() {
+        return this.items;
+    }
```
Comment (damonkelley) [inline src/main/java/com/example/gildedrose/item/GildedRose.java]: **question**  Why is it necessary to add a getter to the GildedRose class?

What are some other ways we could accomplish our goal without expanding the interface of this class? Does this violated any SOLID principles?

File: src/main/java/com/example/gildedrose/GildedRoseService.java (hunk 2 above)
Comment (kudukacar) [reply src/main/java/com/example/gildedrose/GildedRoseService.java]: I guess we could just pass the items to GildedRoseService?

File: src/main/java/com/example/gildedrose/item/GildedRose.java (hunk 3 above)
Comment (kudukacar) [reply src/main/java/com/example/gildedrose/item/GildedRose.java]: I suppose we don't need the getter if we make items public and pass the items to GildedRoseService.

Comment (martingaston) [reply src/main/java/com/example/gildedrose/item/GildedRose.java]: **thought**: Would making `items` public expose any risks down the line? Years from now could I, as a new developer on the project with a big deadline, cause accidental chaos with something like the below? 

`GildedRose.items = new ArrayList<Updateable>(Arrays.asList(new IHaveABigDeadlineTodayItem()))` 

There's perhaps some opportunities for encapsulation here 😸

File: .travis.yml [hunk 4]
```diff
@@ -7,6 +7,8 @@ cache:
     - $HOME/.gradle/caches/
     - $HOME/.gradle/wrapper/
 deploy:
+  on:
+    branch: master
```
Comment (martingaston) [inline .travis.yml:11]: **thought (non-blocking):** I think it's _usually_ a good idea to keep pipeline changes in a separate PR, so they can be isolated if necessary. I do not always follow this advice to the letter myself, I do occasionally regret that though 😄 

There can be other benefits to smaller PRs as well!

File: src/main/java/com/example/gildedrose/service/ItemRepository.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package com.example.gildedrose.service;
+
+import com.example.gildedrose.item.Updateable;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+
+@Service
+public class ItemRepository {
+    private ArrayList<Updateable> items;
```
Comment (paytonrules) [inline src/main/java/com/example/gildedrose/service/ItemRepository.java:10]: At the risk of being silly - at some point you'll probably want to look up one item at a time. Will each item have an ID? It could just be the index into the Array I suppose.
